src/Project/EventListener/ProjectPostUpdateNotifier.php

```diff
@@ -14,17 +15,21 @@ class ProjectPostUpdateNotifier
 {
   protected AchievementManager $achievement_manager;
   protected TagRepository $tag_repository;
+  private ProjectMachineTranslationRepository $machine_translation_repository;
 
-  public function __construct(AchievementManager $achievement_manager, TagRepository $tag_repository)
+  public function __construct(AchievementManager $achievement_manager, TagRepository $tag_repository,
+                              ProjectMachineTranslationRepository $machine_translation_repository)
   {
     $this->achievement_manager = $achievement_manager;
     $this->tag_repository = $tag_repository;
+    $this->machine_translation_repository = $machine_translation_repository;
   }
 
   public function postUpdate(Program $project, LifecycleEventArgs $event): void
```

Comment:
I'm assuming this postUpdate method gets called after there is an update to the database or something so the original API call to update the project can return and this update will happen later after the API call has returned?

Reply:
postUpdate is a doctrine event, see definition here https://www.doctrine-project.org/projects/doctrine-orm/en/2.11/reference/events.html#reference-events-post-update-remove-persist
i don't think this is called after the api has already returned. I think it's fine because you can argue that invalidating the cache is an important part of executing such apis, so if invalidating fails, the api should fail.